src/charm.py

```diff
@@ -143,17 +226,32 @@ def _downstream_prometheus_scrape_relation_broken(self, _):
         self._stored.have_prometheus = False
         self._set_status()
 
+    def _forward_nrpe(self, _):
+        """Send NRPE jobs over to MetricsEndpointAggregator."""
+        nrpes = self.nrpe_exporter.endpoints()
+
+        for nrpe in nrpes:
+            self.metrics_aggregator.set_target_job_data(
+                nrpe["target"], nrpe["app_name"], nrpe["additional_fields"]
+            )
+
     def _set_status(self):
         message = ""
         if (self._stored.have_grafana and not self._stored.have_dashboards) or (
             self._stored.have_dashboards and not self._stored.have_grafana
         ):
             message = " one of (Grafana|dashboard) relation(s) "
 
-        if (self._stored.have_prometheus and not self._stored.have_targets) or (
-            self._stored.have_targets and not self._stored.have_prometheus
+        if (
+            self._stored.have_prometheus
+            and not (self._stored.have_targets or self._stored.have_nrpe)
+        ) or (
+            (self._stored.have_targets or self._stored.have_nrpe)
+            and not self._stored.have_prometheus
         ):
-            message += "{} one of (Prometheus|target) relation(s)".format("and" if message else "")
+            message += "{} one of (Prometheus|target|nrpe) relation(s)".format(
```

Comment:
We owe to the Juju admin to be more specific than this. But I would be OK this being in a follow-up issue+PR.

Reply:
I'll see if I can think of something both short enough to fit into a status message yet descriptive enough.

Otherwise, we log a better message

Reply:
Logs are way too hidden in Juju, we should not assume that people have, like us, `juju debug-log` on the whole time.